### PR TITLE
Widespread the use of pathlib.Path

### PR DESCRIPTION
**Pull request by wizmer**
- neurom.io.utils `get_files_by_path` and `get_morph_files` returns pathlib objects
- fix name == None when calling load_neuron with a pathlib object
- all unit tests use pathlib instead of os.path

### CODE REVIEW

File: neurom/check/runner.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def run(self, path):
 
         for _f in utils.get_files_by_path(path):
             L.info(SEPARATOR)
-            status, summ = self._check_file(_f)
+            status, summ = self._check_file(str(_f))
```
Comment (mgeplf) [inline neurom/check/runner.py:61]: _check_file should take a Path?

File: neurom/check/tests/test_runner.py [hunk 2]
```diff
@@ -76,7 +77,7 @@
 
 REF_0 = {
     'files': {
-        NRN_PATH_0: {
+        str(NRN_PATH_0): {
```
Comment (mgeplf) [inline neurom/check/tests/test_runner.py:80]: not sure it's worth changing to Path here

File: neurom/io/utils.py [hunk 3]
```diff
@@ -94,30 +94,31 @@ def get_morph_files(directory):
     Returns:
         list with all files with extensions '.swc' , 'h5' or '.asc' (case insensitive)
     """
-    lsdir = (os.path.join(directory, m) for m in os.listdir(directory))
-    return list(filter(_is_morphology_file, lsdir))
+    directory = Path(directory)
+    return [Path(directory, m) for m in directory.iterdir() if _is_morphology_file(m)]
```
Comment (mgeplf) [inline neurom/io/utils.py:98]: `iterdir` doesn't return Path items?

File: neurom/check/runner.py (hunk 1 above)
Comment (wizmer) [reply neurom/check/runner.py:61]: Ack

**Comment by asanin-epfl** [comment]
You are still using `os.path` in tests. What's the plan on them?

**Comment by asanin-epfl** [comment]
Here are the occurrences:
Screen Shot 2020-07-21 at 2 05 20 PM

**Comment by wizmer** [comment]
> You are still using `os.path` in tests. What's the plan on them?

Ah yes, my find and replaced froze and did not finish the job at the first try. I forgot.
That should be alright now.

**Comment by wizmer** [comment]
Thank you !

**Pull request merged**